gui_agents/service/config.py

```
@@ -0,0 +1,241 @@
+            # Create config instance
+            config = cls()
+            
+            # Update with file data
+            for key, value in data.items():
+                if hasattr(config, key):
+                    setattr(config, key, value)
+            
+            return config
+            
+        except json.JSONDecodeError as e:
+            raise ConfigurationError(f"Invalid JSON in config file: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)